tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1090,6 +1109,107 @@ void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* it
     }
 }
 
+void CorProfiler::InitializeTraceMethods(WCHAR* id, WCHAR* dd_trace_methods)
+{
+    shared::WSTRING definitionsId = shared::WSTRING(id);
+    std::scoped_lock<std::mutex> definitionsLock(definitions_ids_lock_);
+
+    if (definitions_ids_.find(definitionsId) == definitions_ids_.end())
+    {
+        Logger::Info("InitializeTraceMethods: Id not processed, exiting.");
+        return;
+    }
+
+    if (trace_methods_integration_type_ == nullptr)
+    {
+        Logger::Warn("InitializeTraceMethods: TraceMethodIntegration not initialized, exiting.");
+        return;
+    }
+
+    // TODO we do a handful of string splits here. We could probably do this with indexOf operations instead, but I'm gonna
+    // first make sure this works
+    if (rejit_handler != nullptr)
+    {
+        std::vector<IntegrationDefinition> integrationDefinitions;
+
+        shared::WSTRING dd_trace_methods_string = shared::WSTRING(dd_trace_methods);
+        auto dd_trace_methods_type = shared::Split(dd_trace_methods_string, ';');
+
+        for (const shared::WSTRING& trace_method_type : dd_trace_methods_type)
```

Comment:
Is it possible to add utests for this logic?

Reply:
I moved some of the logic and added unit tests in 14a7c2fd83c471c9c45c5cddf01f1d43328c9219